apps/site/assets/ts/stop/components/DeparturesAndMap.tsx

```diff
@@ -52,6 +68,33 @@ const DeparturesAndMap = ({
     setDepartureFilters(filters);
   };
 
+  useEffect(() => {
+    console.log(departureFilters);
+    if (departureFilters && departureFilters.route) {
+      // grab all the alerts that are current or upcoming
+      const currentAndUpcomingAlerts = filter(alerts, a =>
+        isUpcomingOrCurrentLifecycle(a)
+      );
```

Comment:
question: Are the alerts that we get from the server not already all upcoming or current? Would we want that at the top level so that we don't have to re-do this step?

Reply:
Alerts is a confusing beast.  There appear to be multiple fields that imply the same state which has been making the logic parsing confusing.  `lifecycle` and the `active_period` seem to imply the same state.  The components above filter the alerts on `active_period`.  This here is filtering on `lifecycle`.  Is it guaranteed that `lifecycle` and `active_period` will reflect the same information?

I do think this function call isn't really needed.  I added it because the ticket said specifically only current, and upcoming alerts should be displayed.  Figured best to cover bases. If `active_period` filtering will match `lifecycle` then it is not needed.

Reply:
Ahh fair points.

> Is it guaranteed that `lifecycle` and `active_period` will reflect the same information?

I'm not sure about _guaranteed_, but it feels like a reasonable assumption to me. Still, I am not opposed to leaving this safeguard in.

Thanks for elaborating.